### PR TITLE
Nc news 07

### PR DESCRIPTION
**Pull request by crissu91**

### CODE REVIEW

**Review by Beth-McCormack — CHANGES_REQUESTED** [review]
This is looking good so far! 
Please could you add some visual feedback for the user for if the patch request fails, as currently the screen goes white. 
I also dont think the error handling you have is working how you expect - currently its `Cannot read properties of undefined (reading 'status')`
Please have a go at implementing this and submit another PR 🙂

File: src/components/SingleArticle.jsx [hunk 1]
```diff
@@ -1,37 +1,67 @@
...
+            }))
+        })
+    }
+    
     if (isLoading) {
         return <p>Loading...</p>
     }
 
+    if (apiError) {
+        return (
```
Comment (NRMSMITH) [inline src/components/SingleArticle.jsx:48]: You've made the effort to decrement the votes, but you're actually returning a different component and therefore not showing the decrement.

How about conditionally rendering a p tag that suggests they should try again?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Hey! Functionally, this is working, but it's a bit jarring for the user - I think you need to readdress how you display the error to the user. Have a look at my comment inside :)

Once you've done this, please resubmit. 😄 🚀

File: src/components/SingleArticle.jsx [hunk 2]
```diff
@@ -29,8 +30,11 @@ function SingleArticle() {
         setArticleVotes((currentArticleVotes=>{
             return currentArticleVotes + 1;
         }))
-        patchArticleById(article_id, 1).catch((err)=>{
-            setApiError(err)
+        patchArticleById(article_id, 'a').catch((err)=>{
+
```
Comment (xViolaine) [inline src/components/SingleArticle.jsx]: Your voting system currently allows user to Upvote an Article multiple times. The button isn't disabled between votes, allowing users to inflate votes by spam-clicking the button - this is not false per se, but just something for you to consider :)

File: src/components/SingleArticle.jsx [hunk 3]
```diff
@@ -40,6 +44,7 @@ function SingleArticle() {
     if (isLoading) {
         return <p>Loading...</p>
     }
+
```
Comment (xViolaine) [inline src/components/SingleArticle.jsx:45]: The error works fine!

**Review by xViolaine — APPROVED** [review]

File: src/components/Error.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const Error = ({errorStatus, errorMessage}) => {
+    return (
+        <div className="error">
```
Comment (RichardNeat) [inline src/components/Error.jsx:3]: Could we use anything better than a div here?

**Review by RichardNeat — COMMENTED** [review]
Nice one Cristina - just one comment below to take a look at but otherwise you're good to merge :)

**Pull request merged**